Entity/Components/EnergyComponent.gd

```diff
@@ -23,20 +23,20 @@ func use_energy(card: CardBase) -> void:
 	energy -= card.energy_cost
 
 	if energy < 0:
-		energy = 0
+		push_error("energy cannot be negative")
 
 	on_energy_changed.emit()
 
 
 func on_turn_end() -> void:
-	if energy >= max_energy:
-		energy = max_energy
-		
-	else:
-		energy += energy_generation
-	
+
+	energy += energy_generation
+
+	if energy >= Max_energy:
+		energy = Max_energy
+
 	on_energy_changed.emit()
 
 func add_energy(value: int) -> void:
 	energy += value
-	print(energy)
\ No newline at end of file
+	print(energy)
```

Comment:
is there a reason for printing the energy ?

Reply:
I forgot to delete the print.